From 063389d65f1b42abe672add1b01a0ebe8d2adb6e Mon Sep 17 00:00:00 2001
From: klzgrad <kizdiv@gmail.com>
Date: Sun, 16 May 2021 20:57:51 +0800
Subject: [PATCH] android: build: Disable Android java templates

Default Android tests to raw executables when Java templates are disabled,
including Chromium builds that only need the native library dependency graph.

---
 src/build_overrides/build.gni | 2 +-
 src/testing/test.gni         | 5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/build_overrides/build.gni b/src/build_overrides/build.gni
index ea3301998..fdf2e9c5b 100644
--- a/src/build_overrides/build.gni
+++ b/src/build_overrides/build.gni
@@ -11,7 +11,7 @@ import("//build/config/gclient_args.gni")
 # default_android_ndk_version = "r10e"
 
 # Some non-Chromium builds don't support building java targets.
-enable_java_templates = true
+enable_java_templates = false
 
 # Variable that can be used to support multiple build scenarios, like when
 # V8 is embedded within a target.
diff --git a/src/testing/test.gni b/src/testing/test.gni
--- a/src/testing/test.gni
+++ b/src/testing/test.gni
@@ -445,8 +445,11 @@ template("test") {
       # do not use the custom wrapper
       # (1) Do not cause "gn gen" to fail
       # (2) Do not need to be moved into if(build_with_chromium) block.
+      # Java-free embedders also need raw native tests when they retain
+      # build_with_chromium for the upstream build configuration.
       _use_raw_android_executable =
-          !build_with_chromium && _use_default_launcher
+          _use_default_launcher &&
+          (!build_with_chromium || !enable_java_templates)
     } else {
       not_needed([ "_use_default_launcher" ])
       _use_raw_android_executable = invoker.use_raw_android_executable
-- 
2.53.0

